frontend/test/metabase/scenarios/native/reproductions/15029-sql-variable-dot.cy.spec.js

```diff
@@ -0,0 +1,25 @@
+import { restore, openNativeEditor } from "__support__/e2e/cypress";
+
+describe("issue 15029", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should allow dots in the variable reference (metabase#15029)", () => {
+    openNativeEditor().type(
+      "select * from products where RATING = {{number.of.stars}}",
+      {
+        parseSpecialCharSequences: false,
+      },
+    );
+
+    cy.findAllByText("Variable name")
+      .parent()
+      .as("variableField");
+
+    cy.get("@variableField")
+      .first()
+      .findByText("number.of.stars");
```

Comment:
Nit: No need to alias it if this is the only place in the whole test we're gonna use it.

If you wanted to do it in order to make code more clear, you can also do a custom function `getVariableField().chainTheRest(...)`

Reply:
Great catch, I've simplified it.